### PR TITLE
how to customize clusterapi component

### PR DESCRIPTION
**Pull request by rhomast**

### CODE REVIEW

**Comment by rhomast** [comment]
I moved the CAPI overview topic since I think it makes more sense under 'provisioning cluster' than 'installing verrazzano'. I also made some edits to the topic to clarify the difference between CAPI (as in the generic kubernetes project) and clusterAPI component

File: content/en/docs/setup/provision-cluster/clusterapi.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+   name: example-verrazzano
+ spec:
+   profile: dev
+   components:
+    clusterAPI:
+        enabled: true
+        overrides:
+        - values:
+            global:
+                registry: my.registry
```
Comment (mgianatagh) [inline content/en/docs/setup/provision-cluster/clusterapi.md:78]: For this example, there is no need to provide a global registry value, because the url values in each of the defaultProviders below would take precedence over the registry value. One instance where global.registry override would be used is if the user wants to only override the registry portion of the default file path.

File: content/en/docs/setup/provision-cluster/clusterapi.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+ spec:
+   profile: dev
+   components:
+    clusterAPI:
+        enabled: true
+        overrides:
+        - values:
+            defaultProviders:
+                ocneBootstrap:
+                    version: 1.0
```
Comment (anders-swanson) [inline content/en/docs/setup/provision-cluster/clusterapi.md:43]: Example should use the current ocne and oci versions for our release

Comment (anders-swanson) [reply content/en/docs/setup/provision-cluster/clusterapi.md:43]: For OCNE, this is v1.6.1. 
For OCI, this is v0.9.0.

File: content/en/docs/setup/provision-cluster/clusterapi.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+   components:
+    clusterAPI:
+        enabled: true
+        overrides:
+        - values:
+            global:
+                registry: my.registry
+            defaultProviders:
+                ocneBootstrap:
+                    url: https://my.private.network/cluster-api-provider-ocne/releases/tag/v1.0.0
```
Comment (anders-swanson) [inline content/en/docs/setup/provision-cluster/clusterapi.md:81]: Again, for an example we should use our shipped versions.

File: content/en/docs/setup/provision-cluster/CAPI.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+---
+
+The Cluster API (CAPI) project was developed as a sub-project of Kubernetes and strives to standardize a set of Kubernetes-style APIs for cluster management. External organizations can then build upon these standard APIs to develop custom cluster management solutions.
+
+Learn more about CAPI at [Kubernetes Cluster API Documentation](https://cluster-api.sigs.k8s.io/introduction.html).
+
+Verrazzano incorporates CAPI functionality through the clusterAPI component, which provides the ability to quickly design and deploy clusters and then continue managing your clusters throughout their life cycle, all from within Verrazzano.
+
+{{% alert title="NOTE" color="primary" %}}
+The terminology around clusters differs between CAPI and Verrazzano though the underlying concepts are the same. What CAPI calls Management and Workload clusters are equivalent to Admin and Managed clusters, respectively, in Verrazzano.
```
Comment (rosemarymarano) [inline content/en/docs/setup/provision-cluster/CAPI.md:16]: -> equivalent to admin and managed clusters (we haven't capitalized these MC terms unless they begin a sentence)

File: content/en/docs/setup/provision-cluster/clusterapi.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+ ```
+</div>
+{{< /clipboard >}}
+
+
+## Use a private registry
+
+If you want to upgrade the clusterAPI providers but your Verrazzano instance is installed in disconnected environment, you can configure the clusterAPI component to retrieve the provider assets from another location, instead of the public repository.
+
+1. Place the provider assets in a location that is accessible by your disconnected Verrazzano environment.
```
Comment (gilbode) [inline content/en/docs/setup/provision-cluster/clusterapi.md:57]: Do we need to be more clear on what "provider assets" the user needs to put in this location?

File: content/en/docs/setup/provision-cluster/clusterapi.md (hunk 3 above)
Comment (gilbode) [reply content/en/docs/setup/provision-cluster/clusterapi.md:81]: We should put the versions in hugo parameters like we do other versions.

File: content/en/docs/setup/provision-cluster/ocne-oci.md [hunk 6]
```diff
@@ -1,7 +1,7 @@
 ---
 title: Create OCNE Clusters running on OCI from Verrazzano
```
Comment (gilbode) [inline content/en/docs/setup/provision-cluster/ocne-oci.md:2]: I find "running on OCI from Verrazzano" to be a bit hard to follow.  Could we drop the "from Verrazzano"?

File: content/en/docs/setup/provision-cluster/ocne-oci.md [hunk 7]
```diff
@@ -1,7 +1,7 @@
...
 description: "Add Oracle Cloud Native Environment self-managed clusters running on OCI to your multicluster environment"
```
Comment (gilbode) [inline content/en/docs/setup/provision-cluster/ocne-oci.md:3]: Again this is a lot of words that are difficult to digest "Oracle Cloud Native Environment self-managed clusters running on OCI".  Can we make this more concise?

File: content/en/docs/setup/provision-cluster/CAPI.md [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+
+* **Infrastructure** providers standardize the host environment by provisioning any infrastructure or computational resources required by the cluster or machine. 
+
+* **Bootstrap** providers streamline the node creation process by converting servers into Kubernetes nodes. 
+
+* **Control plane** providers work with the Kubernetes API to regulate your clusters, ensuring that they always strive toward a desired state. 
+
+Currently, Verrazzano supports the CAPI provider for Oracle Cloud Native Environment (CAPOCNE) which bundles a bootstrap and a control plane provider together and works with the [CAPOCI infrastructure provider](https://github.com/oracle/cluster-api-provider-oci) offered by Oracle Cloud Infrastructure (OCI).
+
+During the setup process, the bootstrap provider converts a cluster into an admin cluster - a Kubernetes cluster that controls any other, subordinate or 'managed' clusters. It generates certificates, starts and manages the creation of additional nodes, and handles the addition of control plane and worker nodes to the cluster.
```
Comment (rosemarymarano) [inline content/en/docs/setup/provision-cluster/CAPI.md:29]: -> or 'managed' clusters (hmmm, we haven't used single quotes to designate literal; are you referring to a code term, VZ managed clusters, or something else?)

File: content/en/docs/setup/provision-cluster/clusterapi.md [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+
+## Upgrade providers
+
+You can upgrade the individual providers that make up the clusterAPI component. This allows you to take advantage of new features in the providers without upgrading your entire Verrazzano installation.
+
+This example customizes the clusterAPI component as follows:
+
+* Sets the `version` of both the OCNE bootstrap provider and the OCNE control plane provider to `1.0`
+
+    **NOTE**: Since the OCNE bootstrap and OCNE control plane providers are bundled together, make sure to set their overrides to the same version or it may lead to unexpected behavior.
```
Comment (rosemarymarano) [inline content/en/docs/setup/provision-cluster/clusterapi.md:21]: -> Because the OCNE bootstrap 
(since/as – don't use to mean for the reason of; use because. Use since only for matters of time, for example, since the last time the program was run.)

File: content/en/docs/setup/provision-cluster/clusterapi.md [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+                oci:
+                    version: v0.10.0
+ ```
+</div>
+{{< /clipboard >}}
+
+
+## Use a private registry
+
+If you want to upgrade the clusterAPI providers but your Verrazzano instance is installed in disconnected environment, you can configure the clusterAPI component to retrieve the provider assets from another location, instead of the public repository.
```
Comment (rosemarymarano) [inline content/en/docs/setup/provision-cluster/clusterapi.md:55]: -> (typo) is installed in a disconnected environment

**Review by rosemarymarano — COMMENTED** [review]
Few minor literary edits

File: content/en/docs/setup/provision-cluster/clusterapi.md [hunk 11]
```diff
@@ -0,0 +1,89 @@
+---
+title: Customize clusterAPI
```
[the remark this answers is not in the document]
Comment (rhomast) [reply content/en/docs/setup/provision-cluster/clusterapi.md:2]: I was trying to differentiate between Cluster API (as in the Kubernetes projects) vs. clusterAPI the vz component

**Review by rosemarymarano — APPROVED** [review]

File: content/en/docs/setup/provision-cluster/clusterapi.md (hunk 5 above)
Comment (rhomast) [reply content/en/docs/setup/provision-cluster/clusterapi.md:57]: Yes, I'm currently waiting on dev for the specifics on what/how to add provider assets.
